### PR TITLE
PS-440_metadata-manipulator-bootstrap

### PR DESCRIPTION
**Pull request by jygaulier**
first commit

### CODE REVIEW

File: bin/dev/composer-install.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  lib/metadata-manipulator-bundle
+  auth/api
+  expose/api
+  notify/api
+  uploader/api
+  databox/api
+)
+
+for a in "${apps[@]}"; do
+  echo "Updating $a..."
```
Comment (4rthem) [inline bin/dev/composer-install.sh:24]:
```suggestion
  echo "Installing $a..."
```

**Review by 4rthem — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -214,3 +214,5 @@ jobs:
           else
             echo "Publish step skipped."
           fi
+#      - name: Setup upterm session
+#        uses: lhotari/action-upterm@v1
```
Comment (4rthem) [inline .github/workflows/ci.yaml:218]: You should configure your IDE to auto add new line at the end of file

File: databox/api/Dockerfile [hunk 3]
```diff
@@ -4,14 +4,14 @@ FROM ${REGISTRY_NAMESPACE}php-fpm-base:${BASE_TAG} AS api-php
 
 # Warm up composer cache for faster builds
 COPY docker/caching/composer.* ./
-RUN composer install --prefer-dist --no-dev --no-progress --no-suggest --classmap-authoritative --no-interaction --no-scripts \
+RUN composer install --prefer-dist --no-dev --no-progress --classmap-authoritative --no-interaction --no-scripts \
```
Comment (4rthem) [inline databox/api/Dockerfile:7]: why removing `--no-suggest`?

File: databox/api/Dockerfile [hunk 4]
```diff
@@ -4,14 +4,14 @@ FROM ${REGISTRY_NAMESPACE}php-fpm-base:${BASE_TAG} AS api-php
...
     && rm -rf vendor composer.*
 # End warm up
 
 COPY . .
 
 RUN mkdir -p var/cache var/logs var/sessions \
-    && composer install --prefer-dist --no-dev --no-progress --no-suggest --classmap-authoritative --no-interaction \
+    && composer install --prefer-dist --no-dev --no-progress --classmap-authoritative --no-interaction \
```
Comment (4rthem) [inline databox/api/Dockerfile:14]: why removing --no-suggest?

File: databox/api/src/Api/DataProvider/RenditionClassCollectionDataProvider.php [hunk 5]
```diff
@@ -26,11 +26,11 @@ public function getCollection(string $resourceClass, string $operationName = nul
     {
         $criteria = [];
         $filters = $context['filters'] ?? [];
-        if (isset($filters['workspaceId'])) {
-            $criteria['workspace'] = $filters['workspaceId'];
-        }
+//        if (isset($filters['workspaceId'])) {
+//            $criteria['workspace'] = $filters['workspaceId'];
+//        }
```
Comment (4rthem) [inline databox/api/src/Api/DataProvider/RenditionClassCollectionDataProvider.php:31]: why?

File: databox/api/tests/MetadataManipulator/ReaderTest.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+namespace App\Tests\MetadataManipulator;
+
+use Alchemy\ApiTest\ApiTestCase;
+use Alchemy\MetadataManipulatorBundle\MetadataManipulator;
+use PHPExiftool\Driver\Metadata\Metadata;
+use PHPExiftool\Driver\Metadata\MetadataBag;
+use PHPExiftool\Driver\TagGroup\IFD0\Artist;
+use PHPExiftool\Reader;
+
+class ReaderTest extends ApiTestCase
```
Comment (4rthem) [inline databox/api/tests/MetadataManipulator/ReaderTest.php:14]: This test should be in the bundle instead.

File: databox/api/tests/MetadataManipulator/ServiceTest.php [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+declare(strict_types=1);
+
+namespace App\Tests\MetadataManipulator;
+
+use Alchemy\ApiTest\ApiTestCase;
+use Alchemy\MetadataManipulatorBundle\MetadataManipulator;
+use PHPExiftool\Driver\TagGroup\IFD0\Artist;
+use PHPExiftool\Driver\TagGroupInterface;
+
+class ServiceTest extends ApiTestCase
```
Comment (4rthem) [inline databox/api/tests/MetadataManipulator/ServiceTest.php:12]: Idem

File: databox/api/tests/MetadataManipulator/WriterTest.php [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+use Alchemy\MetadataManipulatorBundle\MetadataManipulator;
+use PHPExiftool\Driver\Metadata\Metadata;
+use PHPExiftool\Driver\Metadata\MetadataBag;
+use PHPExiftool\Driver\TagGroup\IFD0\Artist;
+use PHPExiftool\Driver\TagGroupInterface;
+use PHPExiftool\Reader;
+use PHPExiftool\Driver\Value\Mono;
+
+
+class WriterTest extends ApiTestCase
```
Comment (4rthem) [inline databox/api/tests/MetadataManipulator/WriterTest.php:17]: Idem

File: databox/client/Dockerfile [hunk 9]
```diff
@@ -16,6 +16,7 @@ COPY --from=client-build /usr/src/app/build /var/app
 COPY docker/nginx/conf.d /etc/nginx/conf.d
 
 RUN apk add --no-cache libstdc++ \
+    && apk --no-cache add perl \
```
Comment (4rthem) [inline databox/client/Dockerfile:19]: Why did you add `perl` to the client? Did you mean `databox/api/Dockerfile` instead?

**Review by 4rthem — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (jygaulier) [reply .github/workflows/ci.yaml:218]: yep

File: databox/api/tests/MetadataManipulator/ReaderTest.php (hunk 6 above)
Comment (jygaulier) [reply databox/api/tests/MetadataManipulator/ReaderTest.php:14]: ok

File: databox/api/tests/MetadataManipulator/WriterTest.php [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+     */
+    protected function setup(): void
+    {
+        try {
+            $this->service = static::getContainer()->get('metadata-manipulator');
+            $this->reader = $this->service->getReader();
+        }
+        catch(\Exception $e) {
+            // no-op
+        }
```
Comment (4rthem) [inline databox/api/tests/MetadataManipulator/WriterTest.php:44]: Figure it out

File: databox/api/tests/MetadataManipulator/ServiceTest.php [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    private ?MetadataManipulator $service = null;
+
+    protected function setup(): void
+    {
+        try {
+            $this->service = static::getContainer()->get('metadata-manipulator');
+        }
+        catch(\Exception $e) {
+            // no-op;
+        }
```
Comment (4rthem) [inline databox/api/tests/MetadataManipulator/ServiceTest.php:23]: Figure it out

File: databox/api/tests/MetadataManipulator/ReaderTest.php [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    protected function setup(): void
+    {
+        try {
+            $this->service = static::getContainer()->get('metadata-manipulator');
+            $this->reader = $this->service->getReader();
+        }
+        catch(\Exception $e) {
+            // no-op
+        }
```
Comment (4rthem) [inline databox/api/tests/MetadataManipulator/ReaderTest.php:41]: Figure it out

**Review by 4rthem — CHANGES_REQUESTED** [review]
run cs fixer

**Comment by jygaulier** [comment]
Closed in favor of [PS-440_metadata-manipulator](https://github.com/alchemy-fr/phrasea/pull/241) (squashed)
